hive/src/test/java/org/apache/iceberg/hive/TestHiveCatalog.java

```diff
@@ -0,0 +1,151 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.hive;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.apache.hadoop.hive.metastore.api.Database;
+import org.apache.iceberg.AssertHelpers;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.exceptions.NoSuchNamespaceException;
+import org.apache.thrift.TException;
+import org.junit.Assert;
+import org.junit.Test;
+
+
+public class TestHiveCatalog extends HiveMetastoreTest {
+  private static final String hiveLocalDir = "file:/tmp/hive/" + UUID.randomUUID().toString();
+  private static Map meta = new HashMap<String, String>() {
+    {
+      put("owner", "apache");
+      put("group", "iceberg");
+      put("comment", "iceberg  hiveCatalog test");
+    }
+  };
+
+  @Test
+  public void testCreateNamespace() throws TException {
+    Namespace namespace1 = Namespace.of("noLocation");
+    catalog.createNamespace(namespace1, meta);
+    Database database1 = metastoreClient.getDatabase(namespace1.toString());
+
+    Assert.assertTrue(database1.getParameters().get("owner").equals("apache"));
+    Assert.assertTrue(database1.getParameters().get("group").equals("iceberg"));
+
+    Assert.assertEquals("There no same location for db and namespace",
+        database1.getLocationUri(), defaultUri(namespace1));
+
+    AssertHelpers.assertThrows("Should fail to create when namespace already exist" + namespace1.toString(),
+        org.apache.iceberg.exceptions.AlreadyExistsException.class,
+        "Namespace '" + namespace1.toString() + "' already exists!", () -> {
+          catalog.createNamespace(namespace1);
+        });
+
+    meta.put("location", hiveLocalDir);
+    Namespace namespace2 = Namespace.of("haveLocation");
+    catalog.createNamespace(namespace2, meta);
+    Database database2 = metastoreClient.getDatabase(namespace2.toString());
+    Assert.assertEquals("There no same location for db and namespace",
+        database2.getLocationUri(), hiveLocalDir);
+  }
+
+  @Test
+  public void testListNamespace() throws TException {
+    List<Namespace> namespaces;
+    Namespace namespace1 = Namespace.of("dbname1");
+    catalog.createNamespace(namespace1, meta);
+    namespaces = catalog.listNamespaces(namespace1);
+    Assert.assertTrue("Hive db not hive the namespace 'dbname1'", namespaces.contains(namespace1));
```

Comment:
Listing namespaces under `namespace1` should not include `namespace1`, the semantic seems not correct.